core/src/main/java/io/aiven/klaw/config/MigrationUtility.java

```diff
@@ -0,0 +1,199 @@
+package io.aiven.klaw.config;
+
+import io.aiven.klaw.dao.DataVersion;
+import io.aiven.klaw.dao.migration.DataMigration;
+import io.aiven.klaw.dao.migration.MigrationRunner;
+import io.aiven.klaw.error.KlawDataMigrationException;
+import io.aiven.klaw.repository.DataVersionRepo;
+import jakarta.annotation.PostConstruct;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.sql.Timestamp;
+import java.time.Instant;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.tuple.Pair;
+import org.reflections.Reflections;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.annotation.Configuration;
+
+@Configuration
+@Slf4j
+public class MigrationUtility {
+
+  public static final int SUPPORTED_KLAW_VERSION_NUMBER_SYSTEM = 3;
+
+  @Value("${klaw.data.migration.packageToScan:io.aiven.klaw.dao.migration}")
+  private String packageToScan;
+
+  @Value("${klaw.version}")
+  private String currentKlawVersion;
+
+  @Autowired private DataVersionRepo versionRepo;
+
+  @Autowired private ApplicationContext context;
+
+  @PostConstruct
```

Comment:
We can use InitializingBean instead of Postconstruct ?

Reply:
Yeah I don't have a preference on which we use so happy to use initialising Bean instead.